src/lib/state.ts

```diff
@@ -203,6 +215,23 @@ function createState<T>(): InternalStateType<T> {
       });
     },
 
+    requireEquals(state: T) {
+      if (this._contract === undefined)
+        throw Error(
+          'requireEquals can only be called when the State is assigned to a SmartContract @state.'
+        );
+      let layout = getLayoutPosition(this._contract);
+      let stateAsFields = this._contract.stateType.toFields(state);
+      let accountUpdate = this._contract.instance.self;
+      stateAsFields.forEach((x, i) => {
+        AccountUpdate.assertEquals(
+          accountUpdate.body.preconditions.account.state[layout.offset + i],
+          x
+        );
+      });
+      this._contract.wasConstrained = true;
+    },
+
     assertEquals(state: T) {
       if (this._contract === undefined)
```

Comment:
It would be nice if you could get rid of the code duplication, by making the `assert*` methods just call the `require*` versions. Something like
```ts
assertEquals(state: T) {
   return this.requireEquals(state);
}
```
etc

Reply:
Hi @mitschabaude ,

As discussed above I have removed the duplicate code by making assert* call the require* version. 
As well as build the project locally and adding screenshots of it for your reference.
![State_ts_deprecated](https://github.com/o1-labs/o1js/assets/55076843/636f0899-a466-4fec-bc32-b12b8e2421e8)
![State_ts_deprecated_01](https://github.com/o1-labs/o1js/assets/55076843/5792b9dd-1e89-4861-87fd-192ee7145558)


Also, I am making changes to CHANGELOG.md for first time so can you please help to review it. I have added change in Unreleased for now. Let me know if we need move it.

Please let me know if any change is required.

Thank You 😃